common/djangoapps/enrollment/api.py

```diff
@@ -232,6 +232,10 @@ def update_enrollment(user_id, course_id, mode=None, is_active=None):
         msg = u"Course Enrollment not found for user {user} in course {course}".format(user=user_id, course=course_id)
         log.warn(msg)
         raise errors.EnrollmentNotFoundError(msg)
+    else:
+        if mode == 'credit':
```

Comment:
I don't think we need this check here, since we might want to set other attributes for non-credit courses later on.